odoo/addons/base/models/res_users.py

```diff
@@ -552,6 +579,8 @@ def create(self, vals_list):
             if user.partner_id.company_id:
                 user.partner_id.company_id = user.company_id
             user.partner_id.active = user.active
+            if not user.image_1920:
+                user.image_1920 = user._compute_image()
```

Comment:
Maybe the function / method could take the `name` as input and be called before we've even created the object? That would be a bit more complicated (more cases to handle) but it would avoid round-tripping through updates to the user afterwards.